prow/cmd/deck/job_history.go

```diff
@@ -231,12 +232,9 @@ func (bucket blobStorageBucket) listAll(ctx context.Context, prefix string) ([]s
 
 // Gets all build ids for a job.
 func (bucket blobStorageBucket) listBuildIDs(ctx context.Context, root string) ([]int64, error) {
-	ids := []int64{}
+	var ids []int64
 	if strings.HasPrefix(root, logsPrefix) {
 		dirs, err := bucket.listSubDirs(ctx, root)
```

Comment:
Feels like it is against the Go patterns to have an `err` that only stays in scope on like 248 when dirs happens to be `nil` and `err` is not redefined. However I am still getting to know Go!

Reply:
Deferring error checking like this is absolutely not idiomatic go, but unavoidable here, otherwise we will always either get all build ids successful (which for example for the job linked in the issue report takes multiple minutes) or return none. That's why I added the testcase, otherwise the next person will move that code back to where it was.

The redefining/shadowing is indeed an issue though, good catch.

Reply:
Updated to rename the `err` here to `listErr` to avoid the shadowing